modules/dop/providers/projectpipeline/service_impl.go

```diff
@@ -973,21 +960,13 @@ func (p *ProjectPipelineService) ListApp(ctx context.Context, params *pb.ListApp
 			TotalNum:   0,
 		}
 	}
-	timeEnd := time.Now()
-	timeStart := timeEnd.Add(-1 * 24 * time.Hour)
-	for _, v := range pipelineWithAppNames {
-		if _, ok := appNamePipelineNumMap[v.AppName]; !ok {
-			continue
-		}
-		appNamePipelineNumMap[v.AppName].TotalNum++
-		if apistructs.PipelineStatus(v.Status).IsRunningStatus() {
-			appNamePipelineNumMap[v.AppName].RunningNum++
-			continue
-		}
-		if v.StartedAt.AsTime().After(timeStart) &&
-			v.StartedAt.AsTime().Before(timeEnd) &&
-			v.Status == apistructs.PipelineStatusFailed.String() {
-			appNamePipelineNumMap[v.AppName].FailedNum++
+
+	for _, v := range statics.GetPipelineDefinitionStatistics() {
+		appName := getAppNameByRemote(v.Remote)
+		if _, ok := appNamePipelineNumMap[appName]; ok {
+			appNamePipelineNumMap[appName].FailedNum = int(v.FailedNum)
+			appNamePipelineNumMap[appName].RunningNum = int(v.RunningNum)
+			appNamePipelineNumMap[appName].TotalNum = int(v.TotalNum)
```

Comment:
because your value is a pointer, you can simplify like this:
```golang
if v, ok := appNamePipelineNumMap[appName]; ok {
  v.FailedNum = xxx
}
```

Reply:
done